src/domains/heap_tpolyhedra_sympath_domain.h

```diff
@@ -14,37 +14,59 @@ Author: Viktor Malik
 #ifndef CPROVER_2LS_DOMAINS_HEAP_TPOLYHEDRA_SYMPATH_DOMAIN_H
 #define CPROVER_2LS_DOMAINS_HEAP_TPOLYHEDRA_SYMPATH_DOMAIN_H
 
-
-#include "domain.h"
-#include "heap_tpolyhedra_domain.h"
+#include "product_domain.h"
+#include <ssa/local_ssa.h>
 
 class heap_tpolyhedra_sympath_domaint:public domaint
 {
 public:
-  heap_tpolyhedra_domaint heap_tpolyhedra_domain;
+  product_domaint *heap_tpolyhedra_domain;
```

Comment:
:no_entry_sign: Use unique_ptr wherever possible.

Reply:
I refactored all the work with abstract domains, values, and strategy solvers to use `unique_ptr` only. It is implemented as a separate commit.